### DISCUSSION TITLE
Webpack 5 doesn't remove unused code of the imported file

### DISCUSSION THREAD

**Discussion by AlvinLaiPro** [webpack/webpack#14810] [Q&A]
![image](https://user-images.githubusercontent.com/19744126/143159994-dcb06f5a-a1d6-43f5-a161-579cf862bb26.png)
![image](https://user-images.githubusercontent.com/19744126/143159951-32210861-19d3-4241-801b-4360c5aa584e.png)
I cloned the `webpack` repository and build the side-effect example. I add another export variable `test` to the` a.js` in the `big-module-with-flag`.
As I don't use the` test` variable, so I expect `Webpack` will remove it from the bundle. But the fact is the `test` variable is included in the bundle. 
![image](https://user-images.githubusercontent.com/19744126/143160863-5dca094d-164b-4d88-94f7-3873f32cdcae.png)
[README.md](https://github.com/webpack/webpack/files/7592248/README.md)

@sokra , would you mind explaining more details on this? Thanks in advance.

**Comment by nyngwang** [comment]
I don't think you ran it with the flag `--mode production`. Based on the images, your code doesn't get minimized and tree shaking is enabled by default only in production mode(if you want zero-configuration).
